### PR TITLE
Adding orNull

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

File: kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/null.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package io.kotest.property.arbitrary
+
+import io.kotest.property.Arb
+
+fun <A> Arb<A>.orNull(): Arb<A?> = arb(this.edgecases().plus(null as A?)) { rs ->
+   sequence {
+      while (true)
+         yield(if (rs.random.nextBoolean()) null else this@orNull.next(rs))
```
Comment (LeoColman) [inline kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/null.kt]: every other value from the arb will be a null.
Should we consider a more realistic scenario and reduce the amount of nulls generated?

**Comment by sksamuel** [comment]
We had a similar discussion on slack.

File: kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/null.kt (hunk 1 above)
Comment (sksamuel) [reply kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/null.kt]: We had a similar discussion on slack

**Comment by sksamuel** [comment]
Nice.

**Pull request merged**